### DISCUSSION TITLE
Dynamic store creation

### DISCUSSION THREAD

**Discussion by htunnicliff** [ai/nanostores#341] [Q&A] [answered]
Coming from a Recoil background, is there a recommended pattern for using stores that is similar to Recoil's [`atomFamily`](https://recoiljs.org/docs/api-reference/utils/atomFamily)?

**Comment by ai** [comment, marked as answer]
`mapCreator()` returns a function which return stores

https://github.com/nanostores/nanostores/blob/main/map-creator/index.d.ts

**Reply by htunnicliff** [reply]
Oh cool, I didn't know about this. Any idea why it isn't in the docs?

**Reply by ai** [reply]
It is pretty advanced tool and a lot of user are confused

**Reply by thegatesdev** [reply]
I searched on the internet far and wide and couldn't find anything.
Then I decided to check out the discussions 🙏🏼
Please put a reference to this feature somewhere in the docs, it is really useful in some cases.

**Reply by gismya** [reply]
Please write a PR suggestion for how it could be added to the docs, that's a great way of starting to give back to a project 🙂

I'm genuinely curious to see what that would look like, I haven't figured out when one would want to use it.

**Reply by ai** [reply]
I added docs back 9e76971
